### PR TITLE
feat: events viewer

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Implement events viewer.

## Linear Ticket
https://linear.app/autokitteh/issue/UI-507/events-view-vscode-and-saas

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/organisms/events/table.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	const [leftSideWidth] = useResize({ direction: "horizontal", initial: 50, max: 90, min: 10 });
+	const { eventId } = useParams();
+	const { items: sortedEvents, requestSort, sortConfig } = useSort<BaseEvent>(events || []);
+
+	const navigate = useNavigate();
+	const [visibleRows, setVisibleRows] = useState({ start: 0, end: 20 });
+
+	useEffect(() => {
+		fetchEvents();
+	}, [fetchEvents]);
```
Comment (J1za) [inline src/components/organisms/events/table.tsx:37]:
```suggestion
	}, []);
```

File: src/components/organisms/events/table.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+						<div className="mt-4 h-full">
+							<AutoSizer>
+								{({ height, width }) => (
+									<div
+										className="overflow-auto"
+										onScroll={(event) => onScroll(event, height)}
+										style={{ width, height }}
+									>
+										<Table className="relative w-full overflow-visible">
+											<THead>
```
Comment (J1za) [inline src/components/organisms/events/table.tsx:140]: in the table, I see only the header 
image

File: src/components/organisms/topbar/topbar.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+
+export const TitleTopbar = ({ title }: { title: string }) => {
+	return (
+		<div className="flex items-center justify-between gap-5 rounded-b-xl bg-gray-1250 py-3 pl-7 pr-3.5">
+			<div className="relative flex items-end gap-3 font-fira-code text-gray-500">
```
Comment (J1za) [inline src/components/organisms/topbar/topbar.tsx:6]: Don't need this div. move this class to `span` font-fira-code text-gray-500

File: src/components/templates/eventsLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+export const EventsLayout = () => {
+	const { t } = useTranslation("events", { keyPrefix: "topbar" });
+
+	return (
+		<div className="h-screen w-screen pr-5">
+			<div className="flex size-full">
+				<Sidebar />
+
+				<div className="mb-2 flex w-full flex-col transition">
```
Comment (J1za) [inline src/components/templates/eventsLayout.tsx:17]:
```suggestion
				<div className="mb-2 flex w-full flex-col">
```

File: src/components/templates/eventsLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+					<TitleTopbar title={t("title")} />
+
+					<div className="relative flex size-full overflow-hidden rounded-2xl">
+						<Outlet />
+
+						<div className="absolute !-bottom-5 !-right-5">
```
Comment (J1za) [inline src/components/templates/eventsLayout.tsx:23]:
```suggestion
						<div className="absolute -bottom-5 -right-5">
```

File: src/locales/en/services/translation.json [hunk 6]
```diff
@@ -19,6 +19,8 @@
 	"connectionStatusFetchFailExtended": "Failed to fetch the connection status with the error: {{error}}.",
 	"defaultEnvironmentNotFound": "Default environment not found, project ID: {{projectId}}",
 	"issueFetchingDefaultEnvironment": "Couldn't fetch the default environment, project ID: {{projectId}}, error: {{error}}",
+	"fetchFailedForEventList": "Couldn't fetch event list, error: {{error}}",
+	"destinationIsNotConnection": "Event destination is not connection, event ID: {{eventId}}, destinationID: {{destinationId}}",
```
Comment (J1za) [inline src/locales/en/services/translation.json:23]: Where is it used?

File: src/components/organisms/events/table.tsx (hunk 1 above)
Comment (RonenMars) [reply src/components/organisms/events/table.tsx:37]: fixed

File: src/components/organisms/events/table.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/organisms/events/table.tsx:140]: Try to run a few session with webhook for example.
(Manual run doesn't create event)

File: src/components/organisms/topbar/topbar.tsx (hunk 3 above)
Comment (RonenMars) [reply src/components/organisms/topbar/topbar.tsx:6]: fixed

File: src/components/templates/eventsLayout.tsx (hunk 4 above)
Comment (RonenMars) [reply src/components/templates/eventsLayout.tsx:17]: removed

File: src/components/templates/eventsLayout.tsx (hunk 5 above)
Comment (RonenMars) [reply src/components/templates/eventsLayout.tsx:23]: fixed

File: src/locales/en/services/translation.json (hunk 6 above)
Comment (RonenMars) [reply src/locales/en/services/translation.json]: removed

File: src/components/organisms/events/table.tsx [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+										</Table>
+									</div>
+								)}
+							</AutoSizer>
+						</div>
+					) : null}
+				</Frame>
+			</div>
+
+			<Button className={resizeClass} />
```
Comment (J1za) [inline src/components/organisms/events/table.tsx:166]: rebase and use ButtonResize component

File: src/components/organisms/events/table.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/organisms/events/table.tsx:143]: This is how it looks on my side:
image

File: src/components/organisms/events/table.tsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+import { useCacheStore } from "@src/store";
+import { BaseEvent } from "@src/types/models";
+import { cn } from "@src/utilities";
+
+import { Frame, Loader, ResizeButton, TBody, THead, Table, Td, Th, Tr } from "@components/atoms";
+import { SortButton } from "@components/molecules";
+
+import { CatImage } from "@assets/image";
+
+const rowHeight = 40;
```
Comment (J1za) [inline src/components/organisms/events/table.tsx:19]: move to the constants folder

**Review by J1za — APPROVED** [review]

**Pull request merged**